hazelcast/src/test/java/com/hazelcast/json/MapIndexJsonTest.java

```diff
@@ -63,25 +66,29 @@
 public class MapIndexJsonTest extends HazelcastTestSupport {
 
     public static final int OBJECT_COUNT = 1000;
-    private static final String STRING_PREFIX = "s";
+    private static final String STRING_PREFIX = "s🙂榛";
+    private static final String NON_INDEXED_MAP_NAME = "non-indexed-map";
 
     TestHazelcastInstanceFactory factory;
     HazelcastInstance instance;
 
-    @Parameterized.Parameter(0)
+    @Parameterized.Parameter
     public InMemoryFormat inMemoryFormat;
 
     @Parameterized.Parameter(1)
     public MetadataPolicy metadataPolicy;
 
-    @Parameterized.Parameters(name = "inMemoryFormat: {0}, metadataPolicy: {1}")
+    @Parameterized.Parameter(2)
+    public CacheDeserializedValues cacheDeserializedValues;
+
+    @Parameterized.Parameters(name = "inMemoryFormat: {0}, metadataPolicy: {1}, cacheDeserializedValues: {2}")
     public static Collection<Object[]> parameters() {
-        return asList(new Object[][] {
-                {InMemoryFormat.BINARY, MetadataPolicy.OFF},
-                {InMemoryFormat.BINARY, MetadataPolicy.CREATE_ON_UPDATE},
-                {InMemoryFormat.OBJECT, MetadataPolicy.OFF},
-                {InMemoryFormat.OBJECT, MetadataPolicy.CREATE_ON_UPDATE},
-        });
+        return Arrays.stream(InMemoryFormat.values())
```

Comment:
old version without stream api seems more readable to me.

Reply:
Well, I also think that the old version was more readable but right now there are 2 * 2 * 3 possible parameter combinations, so I didn't want to write all of them by hand. But I could revert this change and write all combinations by hand

Reply:
I see then what you did makes sense. But i question to run such a test in regular PR builds with high combinations. Maybe we can extend this test and can create a nightly variant of it to run with these high number of combinations. For regular PR builds we can only have a run with default values of params. WDYT?